icu4c/source/tools/icuexportdata/icuexportdata.cpp

```diff
@@ -710,6 +709,38 @@ UBool permissibleBmpPair(UBool knownToRoundTrip, UChar32 c, UChar32 second) {
     return false;
 }
 
+
+// Find the slice `needle` within `storage` and return its index, failing which,
+// append all elements of `needle` to `storage` and return the index of it at the end.
+template<typename T>
+size_t findOrAppend(std::vector<T>& storage, const UChar32* needle, size_t needle_len) {
+    // bounds check is `i + needle_len <= storage.size()` since the inner
+    // loop will range from `i` to `i + needle_len - 1` (the `-1` is why we use `<=`)
+    for (size_t i = 0; i + needle_len <= storage.size(); i++) {
```

Comment:
```suggestion
    // Last index where we might find the start of the complete needle.
    size_t last_start = storage.size() - needle_len;
    for (size_t i = 0; i <= last_start; i++) {
```

Reply:
Actually, please make that `lastStart` not `last_start`.
And `needleLen` not `needle_len`.
ICU uses camelCase variable names.

Reply:
This isn't going to work because the types are unsigned

Reply:
You are right. I am not used to working with size_t. Elsewhere in ICU we always use int32_t for indexes and lengths, just like int in Java. But if you like size_t for this tools code, it's ok.

Reply:
totally optional:
```
if (needleLen <= storage.size()) {
    size_t lastStart = storage.size() - needleLen;
    for (size_t i = 0; i <= lastStart; i++) {
        ...
```